openquake/risklib/reinsurance.py

```diff
@@ -41,14 +42,72 @@
 DEBUG = False
 
 
-def check_fields(fields, dframe, idxdict, fname):
+def check_fields(fields, dframe, idxdict, fname, policyfname, treaties,
+                 treaty_types):
     """
     :param fields: fields to check (the first field is the primary key)
     :param dframe: DataFrame with the contents of fname
     :param idxdict: dictionary key -> index (starting from 1)
-    :param fname: file containing the fields to check
+    :param fname: file xml containing the fields to check
+    :param policyfname: file csv containing the fields to check
+    :param treaties: treaty names
+    :param treaty_types: treaty types
     """
     key = fields[0]
+    if len(dframe[key].unique()) < len(dframe):
+        raise InvalidFile(f'{policyfname}: {key} contains duplicates')
+    for treaty in treaties:
+        if treaty not in dframe.columns:
+            raise InvalidFile(f'{policyfname}: {treaty} is missing')
+    policies_from_exposure = list(idxdict)[1:]  # discard '?'
+    policies_from_csv = list(dframe.policy)
+    for policy in policies_from_exposure:
+        if policy not in policies_from_csv:
+            raise InvalidFile(f'{policyfname}: policy "{policy}" is missing')
+    if not is_numeric_dtype(dframe.liability):
+        raise InvalidFile(f'{policyfname}: liabilities must be numeric')
```

Comment:
Display the wrong liability and most importantly the line number.

Reply:
I've just removed this check because it was never used, since the check is performed previously, in
https://github.com/gem/oq-engine/pull/8161/files#diff-87536a8c1b0fd5051d8d926d2d410ba5576f265d4535ed6a2c4ebf090b1eb486R887